src/s3/commands/createBucket.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { getLogger } from '../../shared/logger'
+import { S3Node } from '../explorer/s3Nodes'
+import { DefaultCommands, Commands } from '../../shared/vscode/commands'
+import { DefaultWindow, Window } from '../../shared/vscode/window'
+import * as telemetry from '../../shared/telemetry/telemetry'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+
+/**
+ * Creates a bucket in the s3 region represented by the given node.
+ *
+ * Prompts the user for the bucket name, applying basic validations.
+ * Creates the bucket.
+ * Refreshes the node.
+ */
+export async function createBucketCommand(
+    node: S3Node,
+    window: Window = new DefaultWindow(),
+    commands: Commands = new DefaultCommands()
+): Promise<void> {
+    getLogger().debug(`CreateBucket called for ${node}`)
+
+    const bucketName = await window.inputBox.show({
+        prompt: localize('AWS.s3.createBucket.prompt', 'Create Bucket'),
+        placeHolder: localize('AWS.s3.createBucket.placeHolder', 'Bucket Name'),
+        validateInput: validateBucketName,
+    })
+
+    if (!bucketName) {
+        getLogger().info('CreateBucket cancelled')
+        telemetry.recordS3CreateBucket({ result: 'Cancelled' })
+        return
+    }
+
+    getLogger().info(`Creating bucket ${bucketName}`)
+    try {
+        const bucket = await node.createBucket({ bucketName })
+
+        getLogger().info(`Successfully created bucket ${bucket}`)
+        telemetry.recordS3CreateBucket({ result: 'Succeeded' })
+    } catch (e) {
+        getLogger().error(`Failed to create bucket ${bucketName}`, e)
+        window.message.showError(
+            localize('AWS.s3.createBucket.error.general', 'Failed to create bucket {0}', bucketName)
+        )
+        telemetry.recordS3CreateBucket({ result: 'Failed' })
+    }
+
+    await refreshNode(node, commands)
+}
+
+/**
+ * https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html#bucketnamingrules
+ */
+function validateBucketName(name: string): string | undefined {
```

Comment:
Merge with samDeployWizard.validateS3Bucket

Reply:
Yeah I was aware of some other S3 logic but didn't see the duplicated validation. Will merge them.

Reply:
Merged